code/datums/gamemode/role/time_agent.dm

```diff
@@ -0,0 +1,193 @@
+			Now, you are forced to take responsibility for maintaining the time stream by the mysterious 'Time Agency'.<br>\
+			You only have a limited amount of time before this timeline is deemed lost, in which case you will be forcibly extracted and the mission considered a failure.<br>\
+			Locate certain time-sensitive objects scattered around the station, so as to locate the time anomaly and use it for extraction.<br>\
+			This may not be the first time you visit this timeline, and it may not be the last.</span>")
+
+	to_chat(antag.current, "<span class='danger'>Remember that the items you are provided with are largely non-expendable. Try not to lose them, especially the jump charge, as it is your ticket home.</span>")
+
+
+/datum/role/time_agent/ForgeObjectives()
+	AppendObjective(/datum/objective/target/locate/random)
+	if(prob(30))
+		AppendObjective(/datum/objective/target/assassinate)
+	AppendObjective(/datum/objective/freeform/aid)
+
+
+/datum/role/time_agent/process()
+	if(!objectives.GetObjectives().len || locate(/datum/objective/time_agent_extract) in objectives.GetObjectives())
+		return //Not set up yet
+	var/finished = TRUE
+	for(var/datum/objective/O in objectives.GetObjectives())
+		if(!O.IsFulfilled())
+			finished = FALSE
+			break
+	if(finished)
+		to_chat(antag.current, "<span class = 'notice'>Objectives complete. Triangulating anomaly location.</span>")
+		AppendObjective(/datum/objective/time_agent_extract)
+
+/datum/role/time_agent/OnPostSetup()
+	.=..()
+	if(istype(antag.current, /mob/living/carbon/human))
+		var/mob/living/carbon/human/H = antag.current
+		H.delete_all_equipped_items()
+		var/under = new /obj/item/clothing/under/rank/scientist(H)
+		H.equip_to_slot_or_del(under, slot_w_uniform)
+		var/helmet = new /obj/item/clothing/head/helmet/space/time(H)
+		H.equip_to_slot_or_del(helmet, slot_head)
+		var/suit = new /obj/item/clothing/suit/space/time(H)
+		H.equip_to_slot_or_del(suit, slot_wear_suit)
+		var/mask = new /obj/item/clothing/mask/gas/death_commando(H)
+		H.equip_to_slot_or_del(mask, slot_wear_mask)
+		var/camera = new /obj/item/device/chronocapture(H)
+		H.equip_to_slot_or_del(camera, slot_l_store)
+		var/jump_charge = new /obj/item/device/jump_charge(H)
+		H.equip_to_slot_or_del(jump_charge, slot_r_store)
+		var/belt = new /obj/item/weapon/storage/belt/grenade/chrono(H)
+		H.equip_to_slot_or_del(belt, slot_belt)
+		var/gun = new /obj/item/weapon/gun/projectile/automatic/rewind(H)
+		H.put_in_hands(gun)
+		objects_to_delete = list(under, helmet, suit, mask, camera, jump_charge, belt, gun)
+		H.fully_replace_character_name(newname = "John Beckett")
+		H.make_all_robot_parts_organic()
+		var/list/potential_locations = list()
+		for(var/area/maintenance/A in areas)
```

Comment:
spawning in maint without an ID, does he have some way to get out I'm not thinking of?

Reply:
the jump charge. Hit a door with it, and it'll basically remove it for a few seconds